src/main/index.ts

```diff
@@ -76,13 +79,69 @@ app.on('window-all-closed', () => {
   }
 })
 
-ipcMain.on(FileEvents.OpenSketchesDirDialog, async () => {
+ipcMain.handle(DialogEvents.OpenSketchesDirDialog, async () => {
   const result = await dialog.showOpenDialog({
     properties: ['openDirectory'],
   })
 
-  const dirPath = result.filePaths[0]
+  return result.filePaths[0]
+})
+
+ipcMain.handle(
+  DialogEvents.OpenProjectFileDialog,
+  async (): Promise<OpenProjectFileDialogResponse> => {
+    const result = await dialog.showOpenDialog({
+      filters: [{ name: 'Poject', extensions: ['json'] }],
+    })
+
+    if (!result.canceled && result.filePaths.length > 0) {
+      const projectFile = result.filePaths[0]
+
+      try {
+        // Read and parse the JSON from the project file
+        const fileContent = fs.readFileSync(projectFile, { encoding: 'utf8' })
+        const projectData = JSON.parse(fileContent)
+
+        // Get the directory of the project file
+        const projectDir = path.dirname(projectFile)
+
+        // Path to the neighboring 'sketches' directory
+        const sketchesDir = path.join(projectDir, 'sketches')
```

Comment:
Maybe the sketches path(s) are stored in the projectData?

Reply:
So my plan is that the path will definitely be stored in the project data. However I think there should be two possibilities when the user saves the file:
- if upon saving, the sketches dir is in the same directory as the project json even a child directory: save as a relative path
- Otherwise, save it as an absolute path

When loading, we check to see what kind of path it is and locate the folder accordingly. This means that if you're keeping your save files next to your sketches folder, they can be moved around and shared without having to relocate the sketches directory when opening the project in hedron.

But if you save your project somewhere else (e.g. your desktop) and your sketches folder somewhere else, it's still gonna work.